bin/inx-chronicle/src/broker.rs

```diff
@@ -1,52 +1,70 @@
 // Copyright 2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use chronicle::db::MongoDatabase;
+use async_trait::async_trait;
+use chronicle::{
+    db::{MongoDatabase, MongoDbError},
+    runtime::{
+        actor::{context::ActorContext, envelope::HandleEvent, Actor},
+        error::RuntimeError,
+    },
+};
 use log::debug;
-use tokio::sync::mpsc;
+use thiserror::Error;
 
-use crate::listener::InxEvent;
+#[derive(Debug, Error)]
+pub enum BrokerError {
+    #[error(transparent)]
+    RuntimeError(#[from] RuntimeError),
+    #[error(transparent)]
+    MongoDbError(#[from] MongoDbError),
+}
 
+#[derive(Debug)]
 pub struct Broker {
     db: MongoDatabase,
-    receiver: mpsc::UnboundedReceiver<InxEvent>,
 }
 
 impl Broker {
-    pub fn register(db: MongoDatabase) -> BrokerAddr {
-        let (sender, receiver) = mpsc::unbounded_channel();
-        let mut broker = Broker { receiver, db };
-        tokio::spawn(async move { broker.run().await });
-        BrokerAddr { sender }
+    pub fn new(db: MongoDatabase) -> Self {
+        Self { db }
     }
+}
 
-    pub(crate) async fn handle_event(&mut self, event: InxEvent) {
-        match event {
-            InxEvent::Message(message) => {
-                debug!("Received Message Event");
-                self.db.insert_message_raw(message).await.unwrap()
-            }
-            InxEvent::LatestMilestone(milestone) => {
-                debug!("Received Milestone Event");
-                self.db.insert_milestone(milestone).await.unwrap()
-            }
-        }
-    }
+#[async_trait]
+impl Actor for Broker {
+    type Data = ();
```

Comment:
So this `Data` is what the actor produces on initialization, right? Is there maybe a somewhat more descriptive name we could use instead? I find data to be a very vague term, and honestly, something I'ld expect to to constantly produced, not only on initialization, but maybe I am overlooking something. Since `handle_event` allows mutating this data, I wonder if `State` would be more appropriate. To me it makes sense, that the `init` fn would return the initial state of the actor. Wdyt?

Reply:
If we don't have shared data, do we even need that type at all?

Reply:
Feels semantic at best. I consider the state to be the actor's struct. But the difference is more like externally initialized-state vs internally-initialized state.